src/lib.rs

```diff
@@ -236,8 +236,13 @@ pub fn get_writer<'a>(
 pub fn from_path<'a, P: AsRef<Path>>(
     path: P,
 ) -> Result<(Box<dyn io::Read + 'a>, compression::Format), Error> {
-    let readable = io::BufReader::new(std::fs::File::open(path)?);
-    get_reader(Box::new(readable))
+    let path = path.as_ref();
+    let readable: Box<dyn Read> = if path == Path::new("-") {
+        Box::new(io::BufReader::new(io::stdin()))
+    } else {
+        Box::new(io::BufReader::new(std::fs::File::open(path)?))
+    };
+    get_reader(readable)
```

Comment:
Hi, 

Thanks for this great work, but I still maintain that this feature should not be included in niffler. niffler detects the compression format of a stream/file and returns a stream/file in a decompressed format.
This is clearly outside the scope of application. 

Also if a user wants to use a '.' to designate stdin instead of '-' this value should be made customizable.

I understand the interest of this feature but I maintain that it's not at niffler to management it. There are too many corner cases or a choice made in niffler would make it unusable for a user.

Reply:
Fair point, I reverted the changes.